src/js/actions/LibraryActions.js

```diff
@@ -141,21 +120,87 @@ const refresh = () => {
                 if (docs.length === 0) {
                     return utils.getMetadata(filePath);
                 }
-                return docs[0];
+                return null;
             }).then((track) => {
+                if(track === null) return;
                 return app.models.Track.insertAsync(track);
             }).then(() => {
-                const percent = parseInt(addedFiles * 100 / totalFiles);
-                AppActions.settings.refreshProgress(percent);
+                refreshProgress(addedFiles, totalFiles);
                 addedFiles++;
             });
         }, { concurrency: fsConcurrency });
     }).then(() => {
-        AppActions.library.load();
+        AppActions.library.load(); // TODO Reload the library at the end of this
         dispatchEnd();
     }).catch((err) => {
         console.warn(err);
     });
+
+    // TODO progressive loading in the store, don't freeze the app, able to add files/folders when scanning
+};
+
+const refreshProgress = (processed, total) => {
+    store.dispatch({
+        type : AppConstants.APP_LIBRARY_REFRESH_PROGRESS,
+        processed,
+        total,
+    });
+};
+
+const removeFromLibrary = async (tracksIds) => {
```

Comment:
`async` does not give you anything if you're calling sync API.
`async` makes sense where you have some _really_ async function returning a promise or taking a callback.